src/function/builtin.rs

```diff
@@ -267,6 +267,26 @@ pub fn builtin_function(identifier: &str) -> Option<Function> {
         "str::from" => Some(Function::new(|argument| {
             Ok(Value::String(argument.to_string()))
         })),
+        "str::substring" => Some(Function::new(|argument| {
+            let args = argument.as_tuple()?;
```

Comment:
You can use `as_fixed_len_tuple` and then won't need to do bounds checking below.

Reply:
The function can accept either two or three arguments, hence the use of as_tuple.

Reply:
I see, I didn't realise. Nice!